src/instruments/qml/MuseScore/Instruments/internal/InstrumentsTreeItemDelegate.qml

```diff
@@ -245,7 +245,12 @@ Item {
                     return
                 }
 
-                model.itemRole.isVisible = !model.itemRole.isVisible
+                if (!root.isSelected) {
+                    model.itemRole.isVisible = !model.itemRole.isVisible
+                }
+                else {
+                    instrumentTreeModel.setIsVisibleSelectedRows(!model.itemRole.isVisible)
```

Comment:
The delegate does not know about the model it is in. It is an independent component that we can use with another model. Another model may not have a setIsVisibleSelectedRows method and we will have an error.
It is better to make the signal visibilityChangeRequested(isVisible) and process the signal in the InstrumentsPanel

PS: The editor highlights properties(like instrumentTreeModel) that are declared by the parent but not in the current file. Pay attention to it

Reply:
Thanks for the feedback, I believe I did what you suggested
I would appreciate any other feedback you have to give since this is the first time I'm using QT/QML